### ISSUE TITLE
15.1.0 - FSSSignalDiscovered reporting signals in the wrong system

### ISSUE THREAD

**Issue by DarkSession** [EDDiscovery/EDDiscovery#3299]
I would like to raise an issue that was likely introduced with the release of EDDiscovery 15.1.0.

Since yesterday evening (UTC) we started seeing `FSSSignalDiscovered` events sent through the EDDN which reported a number of signals but provided the wrong star system.

The message below reports a number of signals (fleet carriers) being present in the Paestae system, while the correct system is HIP 12099.  

It is likely that the player jumped from HIP 12099 to Paestae while running EDDiscovery. Unfortunately the game sometimes writes the `FSSSignalDiscovered` event before the `FSDJump` event which is why the `SystemAddress` value of the `FSSSignalDiscovered` journal event needs to be considered when collecting those events to make sure that the correct system is used.  

```
{
    "$schemaRef": "https://eddn.edcd.io/schemas/fsssignaldiscovered/1",
    "header": {
        "gatewayTimestamp": "2022-08-16T14:29:10.640800Z",
        "softwareName": "EDDiscovery",
        "softwareVersion": "15.1.0.0",
        "uploaderID": "dbcc57f4247b58748ea1e6d4306136be64d3fe2e"
    },
    "message": {
        "StarPos": [-79.96875, -79.15625, -140.125],
        "StarSystem": "Paestae",
        "SystemAddress": 669880493425,
        "event": "FSSSignalDiscovered",
        "horizons": true,
        "odyssey": true,
        "signals": [{
            "IsStation": true,
            "SignalName": "SERENITY MASC J4J-N5F",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "PEGASUS X0Q-0VN",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "SEIREITEI VHM-42L",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "SWINGING RICHARD'S V3K-4QL",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "COCAINE & BOUNTY HUNT J8W-91B",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "[GPL] Nebuchadnezzar B3X-17Z",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "HMS MOONBEAM H6G-T3M",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "ADYM&EVE H1T-N6F",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "BATTLESTAR PEGASUS T4B-B1Z",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "MOLLY THE MULE L1W-12Z",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "'-II========II=#\"> V1N-W6M",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "KINGSLAYER V5Z-WKY",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "THE THREE SEASHELLS K0H-8QX",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "DRAGONCAVE XLY-WTV",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "AHAHAHAHAHAHAHAHA HZJ-8XQ",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "PLANET DRUIDIA HFV-4VX",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "WICO*KABANOSI K0W-24M",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "Black Site H2X-25X",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "strike systems holding K6K-31K",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "ORION'S ASCENSION  V2J-GKG",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "CASTLE GREYSKULL C[==> H9V-WXF",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "FORTIS FORTUNA ADIUVAT XHX-17B",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "MAELSTROM'S MISTRESS H8Q-97Q",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "SPACEABEGGO V6Y-B4W",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "- SANSTRI - H9V-G1V",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "WOLF MOON JFV-GVZ",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "ADDC-RIP VAN WINKLE- H2H-3TK",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "BLACK MESA ORBITAL XBY-90L",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "GRAYTOPIA V4M-N8X",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "7TH BOBBUGSB.B.C HLT-3QZ",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "THE FLYING DUTCHMAN VHN-T0V",
            "timestamp": "2022-08-16T14:29:09Z"
        }, {
            "IsStation": true,
            "SignalName": "MICHAEL WITTMANN VNH-33Z",
            "timestamp": "2022-08-16T14:29:09Z"
        }],
        "timestamp": "2022-08-16T14:29:09Z"
    }
}
```

**Comment by Eahlstan** [comment]
`2022-08-16T17:55:36.189Z: POST https://eddn.edcd.io:4430/upload/: {"header":{"uploaderID":"Stanhelm","softwareName":"EDDiscovery","softwareVersion":"15.1.0.500"},"$schemaRef":"https://eddn.edcd.io/schemas/fsssignaldiscovered/1","message":{"event":"FSSSignalDiscovered","horizons":true,"odyssey":true,"timestamp":"2022-08-16T17:56:03Z","SystemAddress":9469194610073,"StarSystem":"Baudihe","StarPos":[163.84375,-111.96875,-33.03125],"signals":[{"timestamp":"2022-08-16T17:56:03Z","SignalName":"Redwood Fort"},{"timestamp":"2022-08-16T17:56:03Z","SignalName":"MUD-221 Sagan-class Traveller"},{"timestamp":"2022-08-16T17:56:03Z","SignalName":"Crooked Wood Meadows"},{"timestamp":"2022-08-16T17:56:03Z","SignalName":"Piazzi Vision"},{"timestamp":"2022-08-16T17:56:03Z","SignalName":"Israel Port","IsStation":true},{"timestamp":"2022-08-16T17:56:03Z","SignalName":"Iwamoto Station"},{"timestamp":"2022-08-16T17:56:03Z","SignalName":"$MULTIPLAYER_SCENARIO42_TITLE;"},{"timestamp":"2022-08-16T17:56:03Z","SignalName":"Associated Emigration Administration"}]}}`

`2022-08-16T17:55:42.491Z: POST https://eddn.edcd.io:4430/upload/: {"header":{"uploaderID":"Stanhelm","softwareName":"EDDiscovery","softwareVersion":"15.1.0.500"},"$schemaRef":"https://eddn.edcd.io/schemas/fsssignaldiscovered/1","message":{"event":"FSSSignalDiscovered","horizons":true,"odyssey":true,"timestamp":"2022-08-16T17:56:09Z","SystemAddress":2832832828106,"StarSystem":"Mudrama Kaze","StarPos":[97.9375,-110.5625,-54.1875],"signals":[{"timestamp":"2022-08-16T17:56:09Z","SignalName":"Redwood Fort"},{"timestamp":"2022-08-16T17:56:09Z","SignalName":"MUD-221 Sagan-class Traveller"},{"timestamp":"2022-08-16T17:56:09Z","SignalName":"Crooked Wood Meadows"},{"timestamp":"2022-08-16T17:56:09Z","SignalName":"Piazzi Vision"},{"timestamp":"2022-08-16T17:56:09Z","SignalName":"Israel Port","IsStation":true},{"timestamp":"2022-08-16T17:56:09Z","SignalName":"Iwamoto Station"},{"timestamp":"2022-08-16T17:56:09Z","SignalName":"$MULTIPLAYER_SCENARIO42_TITLE;"},{"timestamp":"2022-08-16T17:56:09Z","SignalName":"Associated Emigration Administration"}]}}`

```
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Redwood Fort" }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"MUD-221 Sagan-class Traveller" }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Crooked Wood Meadows" }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Piazzi Vision" }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Israel Port", "IsStation":true }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Iwamoto Station" }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"$MULTIPLAYER_SCENARIO42_TITLE;", "SignalName_Localised":"Nav Beacon" }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Associated Emigration Administration" }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"FSDJump", "Taxi":false, "Multicrew":false, "StarSystem":"Mudrama Kaze", "SystemAddress":2832832828106, "StarPos":[97.93750,-110.56250,-54.18750], "SystemAllegiance":"Empire", "SystemEconomy":"$economy_HighTech;", "SystemEconomy_Localised":"High Tech", "SystemSecondEconomy":"$economy_Refinery;", "SystemSecondEconomy_Localised":"Refinery", "SystemGovernment":"$government_Feudal;", "SystemGovernment_Localised":"Feudal", "SystemSecurity":"$SYSTEM_SECURITY_high;", "SystemSecurity_Localised":"High Security", "Population":19566758, "Body":"Mudrama Kaze A", "BodyID":2, "BodyType":"Star", "JumpDist":69.233, "FuelUsed":4.986755, "FuelLevel":27.013245, "Factions":[ { "Name":"Mudrama Kaze Empire Assembly", "FactionState":"None", "Government":"Patronage", "Influence":0.120120, "Allegiance":"Empire", "Happiness":"$Faction_HappinessBand2;", "Happiness_Localised":"Happy", "MyReputation":0.000000 }, { "Name":"Aitvaniez Jet Federal Solutions", "FactionState":"None", "Government":"Corporate", "Influence":0.068068, "Allegiance":"Empire", "Happiness":"$Faction_HappinessBand2;", "Happiness_Localised":"Happy", "MyReputation":0.000000 }, { "Name":"Mudrama Kaze Jet Universal Ltd", "FactionState":"None", "Government":"Corporate", "Influence":0.043043, "Allegiance":"Independent", "Happiness":"$Faction_HappinessBand2;", "Happiness_Localised":"Happy", "MyReputation":0.000000 }, { "Name":"Official Mudrama Kaze Law Party", "FactionState":"None", "Government":"Dictatorship", "Influence":0.010010, "Allegiance":"Independent", "Happiness":"$Faction_HappinessBand2;", "Happiness_Localised":"Happy", "MyReputation":0.000000 }, { "Name":"Mudrama Kaze Blue Cartel", "FactionState":"None", "Government":"Anarchy", "Influence":0.010010, "Allegiance":"Independent", "Happiness":"$Faction_HappinessBand2;", "Happiness_Localised":"Happy", "MyReputation":0.000000 }, { "Name":"Revolutionary Party of Mudrama Kaze", "FactionState":"None", "Government":"Democracy", "Influence":0.055055, "Allegiance":"Independent", "Happiness":"$Faction_HappinessBand2;", "Happiness_Localised":"Happy", "MyReputation":0.000000 }, { "Name":"The House of Etherin", "FactionState":"None", "Government":"Feudal", "Influence":0.693694, "Allegiance":"Empire", "Happiness":"$Faction_HappinessBand2;", "Happiness_Localised":"Happy", "MyReputation":0.000000 } ], "SystemFaction":{ "Name":"The House of Etherin" } }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"ShipLocker", "Items":[  ], "Components":[  ], "Consumables":[ { "Name":"healthpack", "Name_Localised":"Medkit", "OwnerID":0, "Count":100 }, { "Name":"energycell", "Name_Localised":"Energy Cell", "OwnerID":0, "Count":100 }, { "Name":"amm_grenade_emp", "Name_Localised":"Shield Disruptor", "OwnerID":0, "Count":99 }, { "Name":"amm_grenade_frag", "Name_Localised":"Frag Grenade", "OwnerID":0, "Count":99 }, { "Name":"amm_grenade_shield", "Name_Localised":"Shield Projector", "OwnerID":0, "Count":99 } ], "Data":[ { "Name":"airqualityreports", "Name_Localised":"Air Quality Reports", "OwnerID":0, "Count":1 } ] }
{ "timestamp":"2022-08-16T17:56:03Z", "event":"Music", "MusicTrack":"Supercruise" }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Redwood Fort" }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"MUD-221 Sagan-class Traveller" }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Crooked Wood Meadows" }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Piazzi Vision" }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Israel Port", "IsStation":true }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Iwamoto Station" }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"$MULTIPLAYER_SCENARIO42_TITLE;", "SignalName_Localised":"Nav Beacon" }
{ "timestamp":"2022-08-16T17:56:09Z", "event":"FSSSignalDiscovered", "SystemAddress":2832832828106, "SignalName":"Associated Emigration Administration" }
```

Although the server has severe hickups right now (multiple events get written multiple times) this shouldn't happen.

**Comment by robbyxp1** [comment]
I think this is a server related problem.  We have check code in there to make sure the system address is the same as the current system before it sends the EDDN entry with the current system name.

Still analysing.  Report if you see any more please.

**Comment by robbyxp1** [comment]
I think the server hiccups are causing EDD to go wrong, will investigate more.

**Comment by robbyxp1** [comment]
Thanks for the report.

**Comment by robbyxp1** [comment]
We have released 15.1.1 to hopefully solve the issue. Thanks for the feedback, invaluable. Rob
